src/components/PastCommitNode.tsx

```diff
@@ -1,114 +1,139 @@
+      if (!branch.tag) {
+        continue;
+      }
+      const tagParts = branch.tag.split("-");
+      const lastTagPart = tagParts[tagParts.length - 1];
+      if (lastTagPart[0] == "g") {
+        const currentIdAbrev = lastTagPart.slice(1);
+        if (currentIdAbrev == idAbrev) {
+          branches.push(branch);
+        }
+      }
     }
-    return this.props.index === -1
-      ? classes(pastCommitWorkingNodeStyle, pastCommitNodeStyle)
-      : classes(pastCommitNodeStyle);
+    return branches;
   }
 
-  getPastCommitLineClass(): string {
-    if (!this.props.isVisible) {
-      return null;
-    }
-    return this.props.isLast ? null : classes(pastCommitLineStyle);
+  expand() {
+    this.setState({expanded: true});
   }
 
-  getContent(): string | number {
-    if (this.props.index === -1) {
-      return 'WORKING';
-    } else if (this.props.index === 0) {
-      return 'HEAD';
-    } else {
-      return this.props.index;
-    }
+  collapse() {
+    this.setState({expanded: false});
   }
 
-  getContentClass(): string {
-    const activeContentStyle =
-      this.props.index === this.props.activeNode
-        ? pastCommitActiveContentStyle
-        : null;
-    if (!this.props.isVisible) {
-      return null;
+  getNodeClass() {
+    if (this.state.expanded) {
+      return classes(pastCommitNodeStyle, pastCommitExpandedStyle)
     }
-    if (this.props.index === -1) {
-      return classes(
-        pastCommitWorkingContentStyle,
-        pastCommitContentStyle,
-        activeContentStyle
-      );
-    } else if (this.props.index === 0) {
-      return classes(
-        pastCommitHeadContentStyle,
-        pastCommitContentStyle,
-        activeContentStyle
-      );
-    } else {
-      return classes(
-        pastCommitNumberContentStyle,
-        pastCommitContentStyle,
-        activeContentStyle
-      );
-    }
-  }
-
-  handleClick(): void {
-    this.props.index === -1
-      ? this.props.setShowList(true)
-      : (this.props.getPastCommit(
-          this.props.pastCommit,
-          this.props.index,
-          this.props.currentFileBrowserPath
-        ),
-        this.props.setShowList(false));
-    this.props.updateActiveNode(this.props.index);
+    return pastCommitNodeStyle
   }
-
   render() {
     return (
-      <div key={this.props.index}>
-        <div
-          className={this.getPastCommitNodeClass()}
-          onClick={() => this.handleClick()}
-        >
-          <span className={this.getContentClass()}>
-            {this.props.isVisible && this.getContent()}
-          </span>
+      <div
+        onClick={() => {!this.state.expanded && this.expand()}}
+        className={this.getNodeClass()}
+      >
+        <div className={pastCommitHeaderStyle}>
+          <div className={pastCommitHeaderItemStyle}>
+            {this.props.pastCommit.author}
+          </div>
+          <div className={pastCommitHeaderItemStyle}>
+            {this.props.pastCommit.commit.slice(0, 9)}
```

Comment:
Is there a reason for showing SHA1 with 9 digits instead of 7 digits which seems to be the default length for short SHA1?

Reply:
I believe there were  9 digits in the mockup. But yeah, makes sense to use 7 instead of 9, since that does seem to be the default.